core/backend/src/IModelDb.ts

```diff
@@ -2752,21 +2758,30 @@ export class SnapshotDb extends IModelDb {
       throw err;
     }
 
+    // unref timer, so it doesn't prevent a process from shutting down.
+    snapshot._restartDefaultTxnTimer = setTimeout(() => {
+      snapshot.restartDefaultTxn();
```

Comment:
this could benefit from a comment explaining why this is necessary.

Reply:
Thanks, I added a comment to the timer defined on snapshotdb explaining it. Let me know your thoughts.